lib/spack/spack/cmd/diy.py

```diff
@@ -50,6 +50,7 @@ def setup_parser(subparser):
 
 
 def diy(self, args):
+    tty.warn("`spack diy` is deprecated in favor of `spack dev-build`.")
```

Comment:
This should really say that all the functionality of *diy* has *moved* to `dev-build`, and that the `diy` command will be removed in a future Spack version.

To remove redundant code, I would recommend that we implement `diy` the same way `spack compilers` is implemented -- that is, delete all the code here and have the parser setup and the code in this method call `dev-build`.  You can just import `spack.cmd.diy` and call its methods.